networkx/algorithms/bipartite/extendability.py

```diff
@@ -0,0 +1,141 @@
+""" Provides a function for computing the extendability of a
+graph which is undirected, simple, connected and bipartite. """
+
+
+import networkx as nx
+from networkx.utils import not_implemented_for
+
+
+__all__ = [
+    "find_extendability",
+]
+
+
+@not_implemented_for("directed")
+@not_implemented_for("multigraph")
+def find_extendability(G):
+    """Computes the extendability of a graph.
+
+    Definition
+    Graph G is $k$-extendable if and only if G has a perfect matching and every
+    set of $k$ independent edges can extend to perfect matching.
+
+    EXTENDABILITY PROBLEM
+    Input: A graph G and a positive integer $k$.
+    Output: Is G $k$-extendable?
+
+    In general case the above problem is co-NP-complete([2]).
+    If graph G is bipartite, then it can be decided in polynomial time([1]).
+
+    The maximization version of the EXTENDABILITY PROBLEM asks to compute the
+    maximum $k$ for which G is $k$-extendable.
```

Comment:
You describe two problems in the doc_string -- both are labeled "extendability", but perhaps should be called "k-extendability" at least for one of them.  Also, you should perhaps have two functions, one for each problem.  Both of them don't need to be in this PR of course, but have you thought about a function `is_k_extendable(G, k)` or something similar?

Please add some words somewhere which specify which problem this function is solving and that the return value is the "k" you refer to in the doc_string.  Also talk about what happens if no such k exists.